core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go

```diff
@@ -96,52 +98,71 @@ func (r *EvmRegistry) feedLookup(ctx context.Context, upkeepResults []EVMAutomat
 			continue
 		}
 
-		feedLookup, err := r.decodeFeedLookup(upkeepResults[i].PerformData)
+		r.lggr.Infof("[FeedLookup] upkeep %s block %d decodeFeedLookup performData=%s", upkeepId, block, hexutil.Encode(upkeepResults[i].PerformData))
+		lookup, err := r.decodeFeedLookup(upkeepResults[i].PerformData)
 		if err != nil {
 			r.lggr.Errorf("[FeedLookup] upkeep %s block %d decodeFeedLookup: %v", upkeepId, block, err)
 			continue
 		}
-		r.lggr.Infof("[FeedLookup] upkeep %s block %d feedLookup=%v", upkeepId, block, feedLookup)
+		lookup.upkeepId = upkeepId
+		lookup.block = block
+		r.lggr.Infof("[FeedLookup] upkeep %s block %d decodeFeedLookup feedKey=%s timeKey=%s feeds=%v time=%s extraData=%s", upkeepId, block, lookup.feedParamKey, lookup.timeParamKey, lookup.feeds, lookup.time, hexutil.Encode(lookup.extraData))
+		lookups[i] = lookup
+	}
 
-		values, retryable, err := r.doMercuryRequest(ctx, feedLookup, upkeepId)
-		if err != nil {
-			r.lggr.Errorf("[FeedLookup] upkeep %s block %d retryable %v doMercuryRequest: %v", upkeepId, block, retryable, err)
-			upkeepResults[i].Retryable = retryable
-			continue
-		}
+	var wg sync.WaitGroup
+	for i, lookup := range lookups {
+		wg.Add(1)
+		r.doLookup(ctx, &wg, lookup, i, upkeepResults)
```

Comment:
goroutine?

Reply:
Will we be writing to `upkeepResults` in parallel? Not sure if we need to add a mutex on top of it

Reply:
two quesions:
* it appears that `r.doLookup()` is blocking, and therefore we don't actually do the lookups in parallel. A go routine should solve that as amir mentioned.
* we only parallelize the the feed lookups on a per-upkeep basis. Would it be possible to parallelize all feed lookups across all upkeeps?

Reply:
fixed the go routine.

re @infiloop2 
I think we are OK with concurrency bc we only access upkeep results at different indices in each go routine so technically we will not write/read the same memory location in two go routines.

re: @RyanRHall 
what this PR tries to do is parallelizing all feed lookups in 1 pipeline run.
if plugin calls this pipeline with several upkeeps (say 10), this will find all the upkeeps which need feed lookup and spawn go routines for all of them.
then in each feed lookup go routine, it will also fetch mercury data for each asset pair requested by user's revert data in parallel. but this is already done in previous PR.